### PR TITLE
feat(modal): resizable modals

### LINKED ISSUE #647: Draggable and resizable modals
**Describe the problem this feature request is solving**
We use modals in our app that are draggable and resizable. Probably won't be able to switch to cashmere modals until this functionality is added, or there's a way to plugin this type of functionality.

**Describe the solution you'd like**
New parameters, `isDraggable` to make the modal draggable by its title bar, and `isResizable` to make it resizable from the bottom right corner.

**Describe alternatives you've considered**
Find this a way to plugin this functionality on my own without forcing this functionality in Cashmere... but seems like lots of apps could benefit from this work.

**If any, which components will have to change, and how**
Modal component. No breaking changes, just new functionality.

**Additional context**
Screenshot from our app, draggable by title, resizable by bottom right corner.
image

### PR DESCRIPTION
**Pull request by andrew-frueh**
adds resizable functionality to modals that works with draggable

closes #647

### CODE REVIEW

File: projects/cashmere/src/lib/sass/modal.scss [hunk 1]
```diff
@@ -207,3 +212,10 @@ $modal-border: 1px solid $gray-200;
     justify-content: center;
 
 }
+
+@mixin hc-modal-resizable {
+    resize: both;
+    overflow: hidden;
+    min-height: 300px;
```
Comment (corykon) [inline projects/cashmere/src/lib/sass/modal.scss:219]: This seems pretty tall for a min height. I've got a modal as short 170px in PB

File: projects/cashmere/src/lib/modal/modal-window.component.spec.ts [hunk 2]
```diff
@@ -42,42 +42,6 @@ describe('hc-modal-window', () => {
         expect(windowComponent._ignoreOverlayClick).toBeTruthy();
     });
 
-    describe('should change size', () => {
```
Comment (corykon) [inline projects/cashmere/src/lib/modal/modal-window.component.spec.ts:45]: Bummer to rip out tests that could probably be fixed. But I'm not enough of a stickler to let this prevent a merge

**Review by corykon — APPROVED** [review]
Looks good to me. Couple minor things I called out

File: projects/cashmere/src/lib/modal/modal-window.component.spec.ts (hunk 2 above)
Comment (andrew-frueh) [reply projects/cashmere/src/lib/modal/modal-window.component.spec.ts:45]: Ugh, I know I thought that same thing - these tests are still relevant and probably shouldn't be removed.  They would just have to be engineered another way.  Maybe you and I can try to figure out a fix to get them back in this afternoon in our 1:1.

File: projects/cashmere/src/lib/sass/modal.scss (hunk 1 above)
Comment (andrew-frueh) [reply projects/cashmere/src/lib/sass/modal.scss]: @corykon I took a look into this one and header/footer both have `min-heights` of 55px and body has a `min-height` of 120px.  So adding those up I brought the resizable min height down to 230px.

File: projects/cashmere/src/lib/modal/modal-window.component.spec.ts (hunk 2 above)
Comment (andrew-frueh) [reply projects/cashmere/src/lib/modal/modal-window.component.spec.ts:45]: Had to start a new set of unit tests for the modal service which didn't exist previously, but allowed me to test the size and resizable options.  Probably some additional coverage could be added on that one, but at least it's a start - and we haven't lost the coverage we had before.

**Review by corykon — APPROVED** [review]
Looks good. Like the revised min-height approach :+1:

**Pull request merged**